### DISCUSSION TITLE
How to write diff for YAML?

### DISCUSSION THREAD

**Discussion by simkimsia** [PrismJS/prism#3323] [Q&A] [answered]
theme: tomorrow night
css file depends on both plugin chosen and theme
version: v1.26.0

chosen languages:
1. YAML
2. Diff
3. Shell Session
4. Bash + Shell

so chosen plugins:

1. [commandline](https://prismjs.com/plugins/command-line/)
2. [line highlight](https://prismjs.com/plugins/line-highlight/)
3. [line numbers](https://prismjs.com/plugins/line-numbers/)
4. [keep markup](https://prismjs.com/plugins/keep-markup/)
5. [diff highlight](https://prismjs.com/plugins/diff-highlight/)

This is the yaml i want to show as git diff
![image](https://user-images.githubusercontent.com/245021/152670211-8f15a4e3-4de7-4a52-973f-03d6968879d9.png)

but i am not exactly sure how to do this despite reading the documentation for diff-highlight.

I understand based on what i read in diff highlight i cannot have a left and right comparison like filemerge in macOS. Which is fine.

But i am also unsure how to write the yaml diff.

For example i am unsure how to generate the highlighted part.

![CleanShot 2022-02-06 at 14 35 23](https://user-images.githubusercontent.com/245021/152670270-7b28ba78-c347-4ef6-81e3-df38a8eb7e63.png)

Is this done by copy pasting from git?

**Comment by RunDevelopment** [comment, marked as answer]
Diff Highlight uses the [unified diff format](https://en.wikipedia.org/wiki/Diff#:~:text=The%20unified%20format). The rules of this format are as follows:

1. Every line starting with a `+` is an addition.
1. Every line starting with a `-` is a deletion.
1. Every line starting with a ` ` (space) is unchanged.
1. Lines starting with `@@` contain range information.

Diff highlight supports the format in full, but you don't have to include range information. It will also work with diffs that don't have any range information. (Frankly, Diff highlight mostly ignores the range infos.)

> how to write the yaml diff

You can do it by hand (see the rules above), but probably want to use a tool like git. It's pretty easy with the `git diff <file>` command. E.g.:

```
PS C:\Users\micha> git diff file.txt
diff --git a/file.txt b/file.txt
index e69de29..2531695 100644
--- a/file.txt
+++ b/file.txt
@@ -0,0 +1,2 @@
 Heading
-some old text
+some new text
```

Just copy'n'paste everything after the range information.

**Reply by simkimsia** [reply]
i need some time tot test this but i am sure it will be correct. once tested will come back to mark as answer, thank you 🙏

**Reply by simkimsia** [reply]
works. thankyou 🙏

For those who googled here

```
<pre class="language-diff-yaml diff-highlight">
    <code>@@ -6,6 +6,7 @@
 services:
   db:
+    platform: linux/amd64
     build: ./docker/production/postgres/
     image: mysite_production_postgres
     volumes:
@@ -16,6 +17,7 @@ services:
     ports:
       - "5432:5432"
   web:
+    platform: linux/amd64
     build:
       context: .
       dockerfile: ./docker/local/django/Dockerfile</code></pre>
```
